src/pr.test.ts

```diff
@@ -0,0 +1,39 @@
+import { ruleFactory } from './index';
+
+describe('AND', () => {
+  it('"and" object rule should short circuit', () => {
+    const rules = [
+      { if: { and: ['foo', 'bar = 42'] }, then: '1' },
+      { return: 'bar' },
+    ];
+
+    expect(ruleFactory(rules)({ foo: false })).toBe(undefined);
+    expect(ruleFactory(rules)({ foo: true })).toBe(42);
+  });
+
+  it('"and" string rule should short circuit', () => {
+    const rules = [{ if: 'foo and bar = 42', then: '1' }, { return: 'bar' }];
```

Comment:
This might need to be a different PR, as the string expression parsing stuff is handled differently than the structural object stuff.